src/main/java/io/vertx/httpproxy/interceptors/impl/BodyInterceptorImpl.java

```diff
@@ -0,0 +1,68 @@
+/*
+ * Copyright (c) 2011-2024 Contributors to the Eclipse Foundation
+ *
+ * This program and the accompanying materials are made available under the
+ * terms of the Eclipse Public License 2.0 which is available at
+ * http://www.eclipse.org/legal/epl-2.0, or the Apache License, Version 2.0
+ * which is available at https://www.apache.org/licenses/LICENSE-2.0.
+ *
+ * SPDX-License-Identifier: EPL-2.0 OR Apache-2.0
+ */
+
+package io.vertx.httpproxy.interceptors.impl;
+
+import io.vertx.core.Future;
+import io.vertx.core.buffer.Buffer;
+import io.vertx.core.json.JsonObject;
+import io.vertx.httpproxy.ProxyContext;
+import io.vertx.httpproxy.ProxyInterceptor;
+import io.vertx.httpproxy.ProxyResponse;
+import io.vertx.httpproxy.Body;
+import io.vertx.httpproxy.impl.BufferingWriteStream;
+import java.util.function.Function;
+
+
+public class BodyInterceptorImpl implements ProxyInterceptor {
+  Function<Buffer, Buffer> modifyRequestBody;
+  Function<Buffer, Buffer> modifyResponseBody;
+
+  private BodyInterceptorImpl() {}
```

Comment:
It's better to make the fields private and final. Unless it's absolutely necessary for testing (and then it's a good practice to add a comment like _open for testing_).